benchmarks/inference-server/vllm/README.md

```diff
@@ -0,0 +1,120 @@
+
+#### Determine number of gpus
+
+`gpu_count` should be configured respective to the size of the model with some
+overhead for the kv cache. Here's an example on figuring out how many GPUs you
+need to run a model:
+
+vLLM defaults to bfloat16 for running supported models on GPUs. For a model with
+dtype of FP16 or bfloat16, each parameter requires 16 bits. A 7 billion
+parameter model requires a minimum of 7 billion * 16 bits = 14 GB of GPU memory.
+A single L4 GPU has 24GB of GPU memory. 1 L4 GPU is sufficient to run the
+`tiiuae/falcon-7b` model with plenty of overhead for the kv cache.
+
+Note that vLLM server supports gpu_count equal to one of 1, 2, 4, or 8.
+
+#### [optional] set-up credentials config with kubeconfig
+
+If you created your cluster with steps from `../../infra/` or with fleet
+management enabled, the existing `credentials_config` must use the fleet host
+credentials like this:
+
+```bash
+credentials_config = {
+  fleet_host = "https://connectgateway.googleapis.com/v1/projects/$PROJECT_NUMBER/locations/global/gkeMemberships/$CLUSTER_NAME"
+}
+```
+
+If you created your own cluster without fleet management enabled, you can use
+your cluster's kubeconfig in the `credentials_config`. You must isolate your
+cluster's kubeconfig from other clusters in the default kube.config file. To do
+this, run the following command:
+
+```bash
+KUBECONFIG=~/.kube/${CLUSTER_NAME}-kube.config gcloud container clusters get-credentials $CLUSTER_NAME --location $CLUSTER_LOCATION
+```
+
+Then update your `terraform.tfvars` `credentials_config` to the following:
+
+```bash
+credentials_config = {
+  kubeconfig = {
+    path = "~/.kube/${CLUSTER_NAME}-kube.config"
+  }
+}
+```
+
+#### [optional] set up secret token in Secret Manager
+
+A model may require a security token to access it. For example, Llama2 from
+HuggingFace is a gated model that requires a
+[user access token](https://huggingface.co/docs/hub/en/security-tokens). If the
+model you want to run does not require this, skip this step.
+
+If you followed steps from `../../infra/`, Secret Manager and the user access
+token should already be set up. If not, it is strongly recommended that you use
+Workload Identity and Secret Manager to access the user access tokens to avoid
+adding a plain text token into the terraform state. To do so, follow the
+instructions for [setting up a secret in Secret Manager here](https://cloud.google.com/kubernetes-engine/docs/tutorials/workload-identity-secrets).
+
+Once complete, you should add these related secret values to your
+`terraform.tfvars`:
+
+```bash
+# ex. "projects/sample-project/secrets/hugging_face_secret"
+hugging_face_secret = $SECRET_ID
+ # ex. 1
+hugging_face_secret_version =  $SECRET_VERSION
+```
+
+### [Optional] Step 2: configure alternative storage
+
+By default, the vLLM yaml spec assumes that the cluster has [local SSD-backed ephemeral storage](https://cloud.google.com/kubernetes-engine/docs/how-to/persistent-volumes/local-ssd)
```

Comment:
Not something required for this PR. Just putting out the thought that we'll want to automate the storage configuration in our inference-server yaml specs in the long run to support multiple storage options.

A note in the documentation is a great start / addition :)

Reply:
Sounds good. Let's take this as a follow up to add. Not sure about adding upcoming things to the doc. I feel like docs are better served showing the currently supported options as is.

Reply:
ah was referring to the current note you added for existing support. No note needed for upcoming things.